popsycle/synthetic.py

```diff
@@ -3512,3 +3527,390 @@ def generate_slurm_scripts(slurm_config_filename, popsycle_config_filename,
+        Default None.
+
+    Outputs
+    -------
+    <output_root2>.h5 : hdf5 file
+        The new .h5 file with PBHs injected in.
+    """
+    ##########
+    # Error handling: check whether files exist and
+    # whether input types are correct.
+    ##########
+
+    if not overwrite:
+        # Check if HDF5 file exists already. If it does, throw an error message
+        # to complain and exit.
+        if os.path.isfile(output_root2 + '.h5'):
+            raise Exception(
+                'That .h5 file name is taken! Either delete the .h5 file, '
+                'or pick a new name.')
+
+    # Error handling/complaining if input types are not right.
+
+    if type(output_root2) != str:
+        raise Exception('output_root must be a string.')
+
+    if type(galaxia_area) != float:
+        if type(galaxia_area) != int:
+            raise Exception('galaxia_area must be a float or an integer.')
+
+    if type(fdm) != float:
+        if type(fdm) != int:
+            raise Exception('fdm must be a float or an integer.')
+
+    if type(pbh_mass) != int:
+        if type(pbh_mass) != float:
+            raise Exception('pbh_mass must be an integer or a float.')
+
+    if type(r_max) != float:
+        if type(r_max) != int:
+            raise Exception('r_max must be a float or an integer.')
+
+    if type(c) != int:
+        if type(c) != float:
+            raise Exception('c must be an integer or a float.')
+
+    if type(r_vir) != float:
+        if type(r_vir) != int:
+            raise Exception('r_vir must be a float or an integer.')
+
+    if seed is not None:
+        if type(seed) != int:
+            raise Exception('seed must be an integer.')
+
+    ##########
+    # Start of code
+    #########
+
+    # Set random seed
+    np.random.seed(seed)
+
+    t0 = time.time()
+
+    #Define parameters for NFW profile calculations
+    r_s = r_vir/c #kpc, scale radius
+    g = 4.3*(10**-3) #(pc*km^2)/(Msun*s^2)
+    h = 70 #km/(s*Mpc)
+
+    #Read in the hdf5 file that doesn't have PBHs. Product of perform_pop_syn.
+    no_pbh_hdf5_file = h5py.File(hdf5_file)
+    key_list = list(no_pbh_hdf5_file)
+    key_list = key_list[:len(key_list)-2]
+    lat_bin = pd.DataFrame(np.array(no_pbh_hdf5_file['lat_bin_edges']))
+    long_bin = pd.DataFrame(np.array(no_pbh_hdf5_file['long_bin_edges']))
+    no_pbh_hdf5_file.close()
+
+    field_of_view_area = 2*((area/np.pi)**(1/2))
+
+    #NFW Profile calculations to determine mass of dark matter within given distance of galactic center
+    rho_crit = ((3*(h**2))/(8*np.pi*g))*(10**6) #Msun/Mpc^3
+    rho_knot = (200/3)*(rho_crit)*((c**3)/(np.log(c+1)-(c/(1+c)))) # Msun/Mpc^3
+    mass_8_kpc = (4*np.pi*rho_knot*(r_s**3)*(np.log((r_s + r_max)/r_s) - (r_max/(r_s+r_max))))*(((10**3)**3)/((10**6)**3))*fdm #Msun
+
+    #Determine the number of PBHs within that distance
+    num_pbh_8kpc = (mass_8_kpc/pbh_mass)
+    num_pbh_8kpc = round(num_pbh_8kpc, -1)
```

Comment:
Why is there a -1 in this round?

Reply:
-1 rounds it to the nearest 10. There were some issues that came up when not rounded to the nearest 10. That was when the code was slightly different. I'll check if it is still needed.